[CS2113T-F14-2] Fitbot

tryyang2001 opened the pull request:
Fitbot is a desktop app that helps university students who are looking to keep track of their fitness and health with the speed and convenience of command-line based tools, especially in times of online school.

tngjunwei commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,360 @@
+Interaction between the classes could be shown by the uml sequence diagram below.
+
+<p align="center" width="100%">
+  <img width="100%" src="images/Architecture.png" alt="Architecture Sequence Diagram"/>
```
There seems to be a operation invocation missing.
![image](https://user-images.githubusercontent.com/11133074/139213659-0cb5a09b-2dbf-436b-b0de-8d9c7ede419a.png)

tngjunwei commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,360 @@
+-When there is an input, the Ui class will retrieve the information from the user.
+-Once the Main class receives the input, it creates a new Parser class to parse the commands.
+-Depends on the method, in this case add food command, main class will execute the command class(not shown)
+based on the command the parser detects.
```
Slight markdown format bug - need to insert space between dash and the starting word

tngjunwei commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,360 @@
+
+<p align="center" width="100%">
+  <img width="100%" src="images/LogicSequenceDiagram.png" alt="Logic Sequence Diagram"/>
+</p>
```
I think this has the same issue as above - missing method invocation.
![image](https://user-images.githubusercontent.com/11133074/139217230-b8116da7-7816-49b4-854e-fa8ab634be6f.png)

tngjunwei replied:
Same issue.
![image](https://user-images.githubusercontent.com/11133074/139217525-0138e571-e683-4ee0-b5b3-5ec09e767265.png)

tngjunwei commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,360 @@
+#### [Proposed] Add a Food Item Feature
+
+![Add Food Item Sequence Diagram](images/AddFoodItemSequenceDiagram.png)
+
```
Perhaps you can replace the alt frame with an opt frame instead? This way, you won't repeat the constructor call.

![image](https://user-images.githubusercontent.com/11133074/139218613-483f85ea-bfc3-4637-912f-19922dc42e4f.png)

tryyang2001 replied:
Sure! The sequence diagram has been updated! Thanks for pointing this out :D

tngjunwei commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,360 @@
+An abstract data type similar to a regular queue or stack data structure in which elements in priority queue are ordered
+and have "priority" associated with each element. The priority can be defined by the coder. In the case of `FoodList`, the
+priority will be defined as earlier date and time will have higher priority.
+(more coming in the future...)
```
Since this is a developer guide, maybe it won't be necessary to include a glossary for these technical terms?

I think this section is more for clarifying the meaning of certain potentially ambiguous terms.

tryyang2001 replied:
OK! We will consider this suggestion :) Thank you!

tngjunwei commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,360 @@
+
+<p align="center" width="100%">
+  <img width="90%" src="images/ProfileClassDiagram.png" alt="Architecture Sequence Diagram"/>
+</p>
```
Perhaps you can list down the objects aggregated in Profile instead? There seems to be a lot of objects, and I am not sure how a class diagram will help here.

tngjunwei commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,360 @@
+<p align="center" width="100%">
+  <img width="90%" src="images/StorageManagerClassDiagram.png" alt="Architecture Sequence Diagram"/>
+</p>
+
```
Similar to the Data component section, perhaps you could list down the interfaces StorageManager implements, etc?

tngjunwei reviewed (commented):
Overall, the developer's guide is very well written. The explanations provided makes it very easy to understand. The minor issues only lie with the diagrams. Keep it up!

Commit:
implement isValid method in Food and Exercise
refactor isValid method
move getActualIndex to ItemList and implement code inheritance there

Commit:
fix spacing for javaDoc in itemlist line 172 to pass check

weidak committed:
Add ProfileUtils class to make profile more OOP

weidak committed:
Add enum class for activity levels

weidak committed:
Merge pull request #177 from tryyang2001/Branch-V2.1

implement isValid method in Food and Exercise

weidak committed:
Merge branch 'master' of https://github.com/AY2122S1-CS2113T-F14-2/tp into v2.1

weidak committed:
Minor change to accessibility in code

tlyi committed:
Merge pull request #178 from weidak/v2.1

Make profile even more OOP

Commit:
update puml diagram add Interface to the class diagram

Commit:
add explanation for why the lifeline does not end even though the object is deleted
update manual testing section (add in instructions for view and delete)

Commit:
update input code example in Add Exercise command

tttyyzzz committed:
Refactor StartState, modify help command, removed \t to avoid confusion, changed \n to LS update userguide accordingly.
Repeat user message on invalid input to prevent raging users.

tttyyzzz committed:
Undo changes that cause test to fail

weidak committed:
Add diagrams for StorageManager and subsequent references

weidak committed:
Update developer guide for loading data

weidak committed:
Remove duplicate picture

weidak committed:
Remove outdated test case for foodlist

tttyyzzz committed:
Add javadoc for classes

Commit:
Merge branch 'master' of https://github.com/AY2122S1-CS2113T-F14-2/tp into Branch-V2.1

Commit:
delete outdated junit test in FoodListTest

tryyang2001 committed:
Merge pull request #183 from tryyang2001/Branch-V2.1

Update DG and Item Bank and Item Class Diagram

tryyang2001 committed:
Merge branch 'master' into v2.1

tttyyzzz committed:
Merge branch 'master' into brach-refactor-startstate

weidak committed:
Merge branch 'master' of https://github.com/AY2122S1-CS2113T-F14-2/tp into v2.1

# Conflicts:
#	src/test/java/seedu/duke/food/FoodListTest.java

weidak committed:
Merge branch 'v2.1' of https://github.com/weidak/tp into v2.1

# Conflicts:
#	src/test/java/seedu/duke/food/FoodListTest.java

tlyi committed:
Merge branch 'master' into v2.1

* master:
  delete outdated junit test in FoodListTest
  update input code example in Add Exercise command
  add explanation for why the lifeline does not end even though the object is deleted update manual testing section (add in instructions for view and delete)
  update puml diagram add Interface to the class diagram
  Minor change to accessibility in code
  Add enum class for activity levels
  Add ProfileUtils class to make profile more OOP
  fix spacing for javaDoc in itemlist line 172 to pass check
  implement isValid method in Food and Exercise refactor isValid method move getActualIndex to ItemList and implement code inheritance there
  Update data integrity of food and exercise
  Fix formatting error in delete command

weidak committed:
Merge pull request #203 from tttyyzzz/brach-refactor-startstate

Refactor StartState

weidak committed:
Merge pull request #208 from weidak/v2.1

Developer guide updates to Storage and Profile

tlyi committed:
Merge commit '0b1fb52c40754a166f48d47632db82fcf219e00b' into v2.1

* commit '0b1fb52c40754a166f48d47632db82fcf219e00b':
  Add javadoc for classes
  Remove outdated test case for foodlist
  Remove duplicate picture
  Update developer guide for loading data
  Add diagrams for StorageManager and subsequent references
  Undo changes that cause test to fail
  Refactor StartState, modify help command, removed \t to avoid confusion, changed \n to LS update userguide accordingly. Repeat user message on invalid input to prevent raging users.

weidak committed:
Set new limits for profile attributes and loading of storage

weidak committed:
Set more limits of profile and storage

weidak committed:
Small updates to docs

xingjie99 committed:
constraint on upcoming exercises
deleting mulitple indexes for upcoming and banks

xingjie99 committed:
UG update

xingjie99 committed:
indentation

xingjie99 committed:
indentation

weidak committed:
Clean up codes in storage and decoders to initialise before reaching main

Commit:
prevent future date input for adding food item

weidak committed:
Merge pull request #212 from weidak/v2.1

Update data validity for profile attributes and storage

Commit:
implement checking for day field to prevent inputs like 31-11-2021, 31-02-2021

Commit:
Merge branch 'master' of https://github.com/AY2122S1-CS2113T-F14-2/tp into Branch-V2.1

Commit:
improve code readability in Food

tlyi committed:
Merge pull request #213 from xingjie99/V2.1

delete multiple items, data validity on upcoming date, update UG

tlyi committed:
Merge pull request #214 from tryyang2001/Branch-V2.1

Bug Fixing

tlyi committed:
Refactor ParserUtil methods to reduce code duplication
Trim inputs so that whitespace do not affect
Add parsing support for multiple indexes
Fix parsing bug for extra/unnecessary paramaters

tlyi committed:
Add parser support for deleting multiple indexes (for ebank, fbank, upcoming)
Make it compulsory for item type to be first word in input after command word
Make it compulsory for prefixes to be separated by whitespace
Refactor Parser and ParserUtil classes to avoid long methods

tlyi committed:
Minor formatting changes

tttyyzzz committed:
New feature -- users are able to exit while creating or repairing profile.

tlyi committed:
Merge pull request #215 from tlyi/v2.1

Fix parser bugs

tttyyzzz committed:
Merge branch 'master' into branch-v2.1
Resolve merge conflict
# Conflicts:
#	src/main/java/seedu/duke/main/CreateName.java

tttyyzzz committed:
pass gradle check

tttyyzzz committed:
Change \n to LS, reduce repeated code

xingjie99 committed:
edited the message for deleting of multiple items at once

tttyyzzz committed:
rename incorrect naming of class diagram

tttyyzzz committed:
Add create profile to developer guide

tttyyzzz committed:
Add in design consideration for create profile.

tttyyzzz committed:
Reduce magic numbers

tttyyzzz committed:
Remove more magic number.

tttyyzzz committed:
Modify Userguide with by letting user know the presence of bye command when creating profile.

tlyi committed:
Merge pull request #217 from xingjie99/editMessages

Messages for deleting multiple items

tlyi committed:
Add some Junit tests for parsing, to add for EditCommandParser next

tlyi committed:
Merge branch 'master' into v2.1

* master:
  Add some Junit tests for parsing, to add for EditCommandParser next

tlyi committed:
Merge branch 'master' of https://github.com/AY2122S1-CS2113T-F14-2/tp

* 'master' of https://github.com/AY2122S1-CS2113T-F14-2/tp:
  edited the message for deleting of multiple items at once

tlyi committed:
Merge branch 'master' into v2.1

* master:
  edited the message for deleting of multiple items at once

tttyyzzz committed:
Merge branch 'master' into branch-v2.1

tryyang2001 committed:
Merge pull request #216 from tttyyzzz/branch-v2.1

Add new functionality of bye when creating profile

xingjie99 committed:
updated deleting multiple items

xingjie99 committed:
pass gradle

Commit:
set upper bound for food and exercise calorie

tlyi committed:
Fix profile bug

tlyi committed:
Merge pull request #218 from xingjie99/editMessages

Delete multiple upcoming exercises and bank items

tlyi committed:
Merge branch 'master' of https://github.com/AY2122S1-CS2113T-F14-2/tp

* 'master' of https://github.com/AY2122S1-CS2113T-F14-2/tp:
  pass gradle
  updated deleting multiple items
  Modify Userguide with by letting user know the presence of bye command when creating profile.
  Remove more magic number.
  Reduce magic numbers
  Add in design consideration for create profile.
  Add create profile to developer guide
  rename incorrect naming of class diagram
  Change \n to LS, reduce repeated code
  pass gradle check
  New feature -- users are able to exit while creating or repairing profile.

tlyi committed:
Merge branch 'v2.1'

* v2.1:
  Fix profile bug

Commit:
fix bug in isWithinSevenDay method

Commit:
Merge branch 'master' of https://github.com/AY2122S1-CS2113T-F14-2/tp into Branch-V2.1

Commit:
check calorie value for all add and edit commands

tlyi committed:
Standardise formatting
Fix some bugs

tryyang2001 committed:
Merge pull request #220 from tryyang2001/Branch-V2.1

Set upper bound for food and exercise calorie

tlyi committed:
Resolve int overflow and storage exceptions

tlyi committed:
Merge branch 'master' of https://github.com/AY2122S1-CS2113T-F14-2/tp

* 'master' of https://github.com/AY2122S1-CS2113T-F14-2/tp:
  check calorie value for all add and edit commands
  fix bug in isWithinSevenDay method
  set upper bound for food and exercise calorie

weidak committed:
Add more lines

xingjie99 committed:
PPP first draft

xingjie99 committed:
commit

tlyi committed:
Fix minor formatting errors
PPP Draft overview

tttyyzzz committed:
Add PPP draft 0

tttyyzzz committed:
Merge branch 'master' into branch-v2.1

tlyi committed:
Checkstyle formatting

Commit:
create PPP draft

tlyi committed:
Merge pull request #222 from tttyyzzz/branch-v2.1

PPP draft

tlyi committed:
Merge pull request #221 from xingjie99/PPP

PPP first draft

tlyi committed:
Merge pull request #223 from tryyang2001/Branch-V2.1

create PPP draft

tlyi committed:
Update AboutUs links

tlyi committed:
Merge pull request #224 from tlyi/v2.1-clean-up

Merge and clean up code

tttyyzzz committed:
Format developers guide create profile section

tttyyzzz committed:
Format developers guide

tttyyzzz committed:
Merge pull request #225 from tttyyzzz/branch-v2.1

Format DG create profile section

weidak committed:
Merge branch 'master' of https://github.com/AY2122S1-CS2113T-F14-2/tp into v2.1

weidak committed:
Fix storage error handling

weidak committed:
Update documentation for PPP and slight changes to DG

weidak committed:
Minor formatting issues

weidak committed:
Merge pull request #226 from weidak/v2.1

Storage bugs are better handled and update to documentations

weidak committed:
Remove redundant error in catch block

xingjie99 committed:
added javadoc

xingjie99 committed:
edit javadoc

xingjie99 committed:
Merge pull request #227 from xingjie99/addJavaDoc

added javadoc

tlyi committed:
Merge branch 'master' into v2.1-clean-up

* master:
  edit javadoc
  added javadoc
  Minor formatting issues
  Update documentation for PPP and slight changes to DG
  Fix storage error handling
  Format developers guide
  Format developers guide create profile section
  create PPP draft
  Add PPP draft 0
  commit
  PPP first draft
  Add more lines

tttyyzzz committed:
Modify developers guide

tttyyzzz committed:
Merge branch 'master' into branch-v2.1

# Conflicts:
#	docs/DeveloperGuide.md
#	src/main/java/seedu/duke/main/CalorieGoalCreator.java

tttyyzzz committed:
Include basic info for PPP

weidak committed:
Merge pull request #228 from tttyyzzz/branch-v2.1

Modify DG and PPP

weidak committed:
Merge branch 'master' of https://github.com/AY2122S1-CS2113T-F14-2/tp into v2.1

Commit:
update PPP, DG and UG

Commit:
Merge branch 'master' of https://github.com/AY2122S1-CS2113T-F14-2/tp into Branch-V2.1

tttyyzzz committed:
Merge branch 'master' into branch-v2.1

tlyi committed:
Refactor Parser code so that every type of item is handled by one method

tlyi committed:
Merge branch 'master' into v2.1-clean-up

* master:
  Include basic info for PPP
  Modify developers guide

weidak committed:
Edit PPP and touch up diagrams

weidak committed:
Minor changes to statements in dg

Commit:
update ppp, delet less important glossary notes, fix incorrect output in adding future exercise

Commit:
update UG output examples for Recording Food and Recording Exercise

weidak committed:
Add instructions for manual testing to developer guide

tlyi committed:
Update instructions for manual testing for Food Bank
Update logic component related diagrams for DG, but explanations to be updated

xingjie99 committed:
PPP

xingjie99 committed:
UG edits

tlyi committed:
Merge pull request #229 from weidak/v2.1

Update documentation

tlyi committed:
Merge branch 'master' into Branch-V2.1

tlyi committed:
Merge pull request #232 from xingjie99/version2.1

XJ V2.1

tlyi committed:
Merge branch 'master' into Branch-V2.1

tlyi committed:
Merge pull request #231 from tryyang2001/Branch-V2.1

Update PPP, DG and UG

tlyi committed:
Review and standardise UG
Fix code bugs

tlyi committed:
Wrong test case

tryyang2001 committed:
Merge pull request #230 from tlyi/v2.1-clean-up

Update UG, DG, bugs

Commit:
restrict the date range to 7 days only (previously was 8)

Commit:
Merge branch 'master' of https://github.com/AY2122S1-CS2113T-F14-2/tp into Branch-V2.1

tttyyzzz committed:
Commit current progress to merge from master

tttyyzzz committed:
Merge branch 'master' into branch-v2.1

tlyi committed:
DG review

tlyi committed:
Merge pull request #233 from tryyang2001/Branch-V2.1

Change the date range to 7 days

tlyi committed:
Merge branch 'master' into v2.1-clean-up

* master:
  restrict the date range to 7 days only (previously was 8)

tlyi committed:
checkstyle error

tryyang2001 committed:
Merge pull request #234 from tlyi/v2.1-clean-up

Review DG

Commit:
update puml diagrams

Commit:
Merge branch 'master' of https://github.com/AY2122S1-CS2113T-F14-2/tp into Branch-V2.1

# Conflicts:
#	docs/diagrams/ItemBankAndItemClassDiagram.puml
#	docs/images/ItemBankAndItemClassDiagram.png

tttyyzzz committed:
Commit to merge branch

tttyyzzz committed:
Merge branch 'master' into branch-v2.1

tttyyzzz committed:
Perform test on Ui and Subclass of AttributeCreator.

weidak committed:
Update implementations

weidak committed:
Merge branch 'master' of https://github.com/AY2122S1-CS2113T-F14-2/tp into v2.1

tttyyzzz committed:
Solidify test and logger

xingjie99 committed:
format message

xingjie99 committed:
edits to UG

tttyyzzz committed:
modify DG and PPP

tttyyzzz committed:
Pass gradle check

xingjie99 committed:
Changes to UG

weidak committed:
Add more sequence diagrams

weidak committed:
Remove extra images

weidak committed:
Add formatting

weidak committed:
Minor format

weidak committed:
Fix DG

tttyyzzz committed:
Add more user stories.

tttyyzzz committed:
Modify Ui sequence diagram and DG

tlyi committed:
Checkstyle errors
Format PPP headers

tlyi committed:
Merge commit '1818af948e74f28aef076379a84734e32c547d06' into v2.1-clean-up

* commit '1818af948e74f28aef076379a84734e32c547d06':

Commit:
update DG diagram and some explanations, change enum to all CAPS, update architecturediagram

xingjie99 committed:
manual testing for exercise items

xingjie99 committed:
delete two  old assertions

weidak committed:
Update class diagrams

weidak committed:
Merge pull request #235 from tttyyzzz/branch-v2.1

Modify DG and update PPP, include logging and test code for startstate and its depended class

weidak committed:
Merge pull request #236 from tlyi/v2.1-clean-up

Fix bugs in delete command and format PPP

tttyyzzz committed:
Merge branch 'master' into branch-v2.1

Commit:
update architecture diagram, add description for State component
add Data Component section in DG
update ItemBankAndItemClassDiagram

Commit:
Merge branch 'master' of https://github.com/AY2122S1-CS2113T-F14-2/tp into Branch-V2.1

# Conflicts:
#	docs/DeveloperGuide.md

xingjie99 committed:
manual testing for DG

weidak committed:
Update old dg

weidak committed:
Merge pull request #237 from xingjie99/FinalV2.1

Update to UG and DG

weidak committed:
Minor fix

Commit:
Merge branch 'master' of https://github.com/AY2122S1-CS2113T-F14-2/tp into Branch-V2.1

weidak committed:
Merge branch 'master' of https://github.com/AY2122S1-CS2113T-F14-2/tp into v2.1

weidak committed:
Add manual testing

Commit:
update description for AddFoodItem seq diagram

weidak committed:
Minor fixes, pass gradle

weidak committed:
Minor update to dg format

weidak committed:
Merge pull request #239 from tryyang2001/Branch-V2.1

Update DG

weidak committed:
Merge branch 'master' of https://github.com/AY2122S1-CS2113T-F14-2/tp into v2.1

xingjie99 committed:
Merge pull request #238 from weidak/v2.1

Update documentation for developer guide

xingjie99 committed:
final edits to PPP UG DG

xingjie99 committed:
Update UserGuide.md

xingjie99 committed:
added photo

Commit:
adjust image size, make header bold

xingjie99 committed:
update picture

tlyi committed:
Final DG and PPP updates

tlyi committed:
Resolve merge conflicts in DG

tlyi committed:
Update picture in AboutUs

tlyi committed:
Add PR link to PPP

tlyi committed:
Merge pull request #240 from xingjie99/finalEdits

Final edits to PPP UG DG

tlyi committed:
Merge pull request #241 from tlyi/v2.1-clean-up

Final edits to DG and PPP

tttyyzzz committed:
Merge remote-tracking branch 'upstream/master' into branch-v2.1

Commit:
update aboutus
make all headers bolded in dg
updates itembank class diagram

weidak committed:
Add data limits

weidak committed:
Merge branch 'master' into v2.1

tlyi committed:
Merge pull request #243 from weidak/v2.1

Add data limits

Commit:
Merge branch 'master' of https://github.com/AY2122S1-CS2113T-F14-2/tp into Branch-V2.1

# Conflicts:
#	docs/AboutUs.md
#	docs/DeveloperGuide.md

Commit:
Merge branch 'master' of https://github.com/AY2122S1-CS2113T-F14-2/tp into Branch-V2.1

Commit:
move Data Limits to section 6
add description to data limits
fix exclamation mark emoji position

tryyang2001 committed:
Merge pull request #242 from tttyyzzz/branch-v2.1

Add photo :)

Commit:
Merge branch 'master' of https://github.com/AY2122S1-CS2113T-F14-2/tp into Branch-V2.1

Commit:
update AboutUs

Commit:
fix typo

Commit:
fix AboutUs

tryyang2001 committed:
Merge pull request #244 from tryyang2001/Branch-V2.1

Updates UG, AboutUs

tlyi committed:
Merge branch 'master' into v2.1-clean-up

* master:
  fix AboutUs
  fix typo
  update AboutUs
  move Data Limits to section 6 add description to data limits fix exclamation mark emoji position
  Add data limits
  update aboutus make all headers bolded in dg updates itembank class diagram
  update picture
  adjust image size, make header bold
  added photo
  Update UserGuide.md
  final edits to PPP UG DG

tttyyzzz committed:
modify format

weidak committed:
Update ppp

tttyyzzz committed:
Merge branch 'master' into branch-v2.1

weidak committed:
Merge branch 'master' of https://github.com/AY2122S1-CS2113T-F14-2/tp into v2.1

# Conflicts:
#	docs/UserGuide.md

tttyyzzz committed:
Make amendments to user stories. Removed fullstops

tttyyzzz committed:
reformat PPP

weidak committed:
Update ppp even more

Commit:
fix typo and formatting issue in DG

tlyi committed:
Update UG and fix code for release

weidak committed:
Merge pull request #245 from tttyyzzz/branch-v2.1

Format PPP

weidak committed:
Merge pull request #246 from tryyang2001/Branch-V2.1

Fix typo and formatting issue in DG

tryyang2001 committed:
Merge pull request #248 from weidak/v2.1

Update PPP

tlyi committed:
Update test cases

weidak committed:
Fix PPP

weidak committed:
Add one line

weidak committed:
Merge pull request #249 from weidak/v2.1

PPP

xingjie99 committed:
Merge pull request #247 from tlyi/v2.1-clean-up

Update UG and fix bugs for release

Commit:
shorten PPP

tlyi committed:
Merge pull request #250 from tryyang2001/Branch-V2.1

Shorten PPP

tlyi committed:
Update DeveloperGuide.md

tlyi committed:
Update DeveloperGuide.md

tlyi committed:
Update UserGuide.md

tlyi committed:
Update UserGuide.md

xingjie99 committed:
format example in UG

tryyang2001 committed:
Merge pull request #251 from xingjie99/UGExamplesFormat

format example in UG

xingjie99 committed:
second update to ug

weidak committed:
Merge pull request #252 from xingjie99/UGExamplesFormat

second update to ug

weidak committed:
Update diagrams

tlyi committed:
Update DeveloperGuide.md

weidak committed:
Modify diagrams

tlyi committed:
Change diagrams colour

weidak committed:
Merge pull request #253 from weidak/v2.1

Update diagrams

weidak committed:
Merge pull request #254 from tlyi/v2.1-clean-up

Change diagrams colour

tlyi committed:
Update UserGuide.md

tlyi committed:
Update UserGuide.md

tlyi committed:
Update UserGuide.md

tttyyzzz committed:
add default calorieGoal if calorieGoal is not initialised

tttyyzzz committed:
fix StatisticsTest

tttyyzzz committed:
Merge pull request #255 from tttyyzzz/post

add default calorieGoal if calorieGoal is not initialised
